core/exception_vectors.S

```diff
@@ -491,7 +530,15 @@ NMIExceptionHandler:
         .balign  4
 UserExceptionHandler:
         .type   UserExceptionHandler, @function
-        xsr     a0, excsave1    # a0 now contains sp
+        // save a0, a1 to debug_saved_ctx before stack pointer is affected
```

Comment:
This allows gdbstub to examine context state before C code for interrupt handling is run, so you can press Ctrl-C in GDB and see where the program is going (like prvIdleTask)